parsers/thueringen.py

```diff
@@ -0,0 +1,199 @@
+from urllib.request import urlopen
+from bs4 import BeautifulSoup as parse
+import re
+from utils import Parser
+from datetime import datetime, timedelta
+from pyopenmensa.feed import LazyBuilder
+
+amount_regex = re.compile('\d{1,2},\d{1,2}')
+
+
+def parse_start_date(document):
+	week_start_end_date_regex = re.compile('\d{1,2}\.\d{1,2}\.\d{4}')
+	calendar_week_regex = re.compile('\d{4}-\d{1,2}$')
+
+	option = document.find("option", value=calendar_week_regex, selected=True)
+
+	if option is None:
+		return None
+
+	start_date_str = week_start_end_date_regex.search(option.text)
+
+	return datetime.strptime(start_date_str.group(), '%d.%m.%Y')
+
+
+def parse_fees(document):
+	fees_regex = re.compile('Bedienstete.*')
+
+	fees = document.find_all('p', class_='MsoNormal', string=fees_regex)
+
+	if len(fees) != 1:
+		return None, None
+
+	fees = fees[0]
+
+	fee_strings = fees_regex.findall(fees.text)
+	for amount_candidate in fee_strings:
+		amount_strings = amount_regex.findall(amount_candidate)
+		if len(amount_strings) != 2:
+			continue
+
+		employees_fee = float(amount_strings[0].replace(',', '.'))
+		guests_fee = float(amount_strings[1].replace(',', '.'))
+		return employees_fee, guests_fee
+
+	return None, None
+
+
+def parse_ingredients(document):
+	contextbox = document.find('div', class_='kontextbox')
+
+	shorts = contextbox.find_all('p')
+
+	digits_regex = re.compile('(\d)+: .*')
+	chars_regex = re.compile('(\w)+: .*')
+
+	digits = dict()
+	chars = dict()
+
+	if len(shorts) < 3:
+		return
+	for short in shorts[1].childGenerator():
+		if digits_regex.search(str(short)):
+			n = str(short).strip('\r\n').split(':')
+			digits[n[0]] = n[1]
+
+	for short in shorts[2].childGenerator():
+		if chars_regex.search(str(short)):
+			c = str(short).strip('\r\n').split(':')
+			chars[c[0]] = c[1]
+
+	shorts = contextbox.find_all('td')
+
+	if len(shorts) >= 1:
+		for short in shorts[1].childGenerator():
+			if chars_regex.search(str(short)):
+				c = str(short).strip('\r\n').split(':')
+				chars[c[0]] = c[1]
+
+	return digits, chars
+
+
+def ingredient_list_to_notes(ingredients_list, chars, digits):
+	ingredients = []
+	for ingredient in ingredients_list:
+
+		if digits.get(ingredient) is not None:
+			ingredients.append(digits[ingredient][1:])
+
+		if chars.get(ingredient) is not None:
+			ingredients.append(chars[ingredient][1:])
+
+	return ingredients
+
+
+def parse_meals(day, digits, chars):
+	ingredients_regex = re.compile('Inhalt:.*')
+
+	meals_t_rows = day.find_all('tr')
+
+	for meal_data in meals_t_rows:
+		meal_t_datas = meal_data.find_all('td')
+		if len(meal_t_datas) != 3:
+			continue
+
+		category = [meal_t_datas[0].text]
+		if len(category) == 1:
+			category = category[0]
+		else:
+			category = 'Hauptgerichte'
+
+		notes = []
+		ingredients = ingredients_regex.findall(meal_t_datas[1].text)
+
+		if len(ingredients) > 0:
+			ingredients_list = ingredients[0].strip('Inhalt: ').strip('\r').split(',')
```

Comment:
Please use `strip()` instead of `strip('\r')`. Also you should strip the resulting list items (maybe combined with the list comprehension from above?), because there may be some spaces in there sometimes.